chia/consensus/blockchain.py

```diff
@@ -312,16 +317,18 @@ async def _reconsider_peak(
                     tx_removals, tx_additions = [], []
                 if block.is_transaction_block():
                     assert block.foliage_transaction_block is not None
-                    await self.coin_store.new_block(
+                    added, _ = await self.coin_store.new_block(
```

Comment:
@Yostra This looks like a bug. `new_block()` returns (*removed*, *added*). here you're binding the removed records to a variable called `added`.

There need to be a test that exercises this and fail as this code is written now. Otherwise we should remove this code to not return these lists at all (since clearly they aren't being used)

Reply:
I agree. Changing the blockchain logic is pretty risky as well, and this can be done externally. We can read the coinstore from outside of this file.

Reply:
Those lists are used only to notify wallet of the changes, in this case (that was broken) change happens when we add a genesis block and in practice this wouldn't happen as no wallet can connect before blockchain is initialized. I'll fix a typo.

Reply:
I think, instead of returning added and remived `CoinRecord`s, we should just return "changed coin IDs". and anything the wallet is subscribed to, we can lookup via the `CoinStore`. Presumably most coins are *not* subscribed to, and it would be a net improvement. That way, we also wouldn't need `new_block()` to return anything, since we already have the coin IDs added and removed at the call site.

Reply:
(I'd be happy to propose that patch when I'm done with what I'm working on right now)